app/controllers/branches_controller.rb

```diff
@@ -34,11 +35,54 @@ def destroy
     redirect_to branches_url
   end
 
+  def meals
+    @branch_meals = BranchMeal.includes(:meal, :branch).where(branch_id: @branch.id).order(:id)
+  end
+
+  def add_meal
+    @branch_meal = BranchMeal.new
+  end
+
+  def create_meal
+    @branch_meal = @branch.branches_meals.build(branch_meal_params)
+
+    if @branch_meal.save
+      redirect_to meals_branch_path(params[:id]), notice: t('.success')
+    else
+      render :add_meal, status: :unprocessable_entity
+    end
+  end
+
+  def toggle_meal_active
+    @branch_meal.update(active: true)
+    redirect_to meals_branch_path(params[:id])
+  end
+
+  def toggle_meal_inactive
+    @branch_meal.update(active: false)
+    redirect_to meals_branch_path(params[:id])
+  end
+
+  def destroy_meal
+    @branch_meal.destroy ? flash[:notice] = t('.success') : flash[:alert] = t('.failure')
```

Comment:
no need to destroy these, we can just deactivate the branch meals

Reply:
Done